covid_xprize/scoring/prescriptor_scoring.py

```diff
@@ -15,21 +17,22 @@ def weight_prescriptions_by_cost(pres_df, cost_df):
 
 
 def generate_cases_and_stringency_for_prescriptions(start_date, end_date, prescription_file, costs_file):
+    start_time = time.time()
     # Load the prescriptions, handling Date and regions
     pres_df = XPrizePredictor.load_original_data(prescription_file)
 
     # Generate predictions for all prescriptions
     predictor = XPrizePredictor()
-    pred_dfs = []
+    pred_dfs = {}
     for idx in pres_df['PrescriptionIndex'].unique():
         idx_df = pres_df[pres_df['PrescriptionIndex'] == idx]
         idx_df = idx_df.drop(columns='PrescriptionIndex')  # Predictor doesn't need this
         # Generate the predictions
         pred_df = predictor.predict_from_df(start_date, end_date, idx_df)
         print(f"Generated predictions for PrescriptionIndex {idx}")
         pred_df['PrescriptionIndex'] = idx
-        pred_dfs.append(pred_df)
-    pred_df = pd.concat(pred_dfs)
+        pred_dfs[idx] = pred_df
```

Comment:
Keeping the generated predictions in a dictionary to map them to their PrescriptionIndex